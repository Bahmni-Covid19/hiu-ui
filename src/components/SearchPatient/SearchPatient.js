import React, { useState, useEffect } from "react";
import SearchPatientStyles from "./SearchPatient.style";
import { IconButton, TextField, CircularProgress } from "@material-ui/core";
import SearchIcon from "@material-ui/icons/Search";

const SearchPatient = ({
  onSearch,
  patientId,
  patientName,
  success,
  loading,
  error,
  serverError
}) => {
  const [localPatientId, setPatientId] = useState(patientId);
  const [textInput, setTextInput] = useState("");
  useEffect(() => {
    if (loading) {
      setPatientId(textInput);
      setTextInput("Searching");
    } else {
      const successText = localPatientId.concat(": ", patientName);
      {
        success ? setTextInput(successText) : setTextInput(localPatientId);
      }
    }
  }, [loading]);

  const generateErrorText = () => {
    if (error) {
      return "Id not found.";
    } else if (serverError) {
      return "Id not found.";
    } else {
      return "";
    }
  };

  return (
    <SearchPatientStyles>
      <div className="search-bar">
        <TextField
          id="search-field"
          disabled={loading}
          error={error}
          helperText={generateErrorText()}
          value={textInput}
          onChange={e => setTextInput(e.target.value)}
        />
        <TextField
          className="fiduciary-text-field"
          disabled
          placeholder="@NCG"
        />
        <IconButton
          disabled={textInput.length == 0 || loading}
          type="button"
          className="icon-button"
          aria-label="search"
          theme="primary"
          onClick={() => onSearch(textInput)}
        >
          {loading ? (
            <CircularProgress
              id="loader"
              className="loader"
              variant="indeterminate"
              disableShrink
              size={24}
              thickness={4}
            />
          ) : (
            <SearchIcon
              id="search-button"
              className="search-button"
              variant="outlined"
              theme="primary"
            />
          )}
        </IconButton>
      </div>
    </SearchPatientStyles>
  );
};

SearchPatient.defaultProps = {
  patientId: "",
  loading: false,
  error: false,
  serverError: false
};

export default SearchPatient;
