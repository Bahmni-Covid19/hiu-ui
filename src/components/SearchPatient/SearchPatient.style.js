import styled from "styled-components";

const SearchPatientStyles = styled.div`
  .root {
    padding: 2px 2px 2px 10px;
    display: flex;
    alignitems: center;
    width: 400;
  }
`;

export default SearchPatientStyles;
