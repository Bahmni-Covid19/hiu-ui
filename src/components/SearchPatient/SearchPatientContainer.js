import React from "react";
import { connect } from "react-redux";
import SearchPatient from "./SearchPatient";
import { onSearch } from "./../../redux/actions/searchAction";

const mapStateToProps = state => ({
  patientId: state.patientDetail.result,
  loading: state.patientDetail.loading
});

const mapDispatchToProps = {
  onSearch
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchPatient);
