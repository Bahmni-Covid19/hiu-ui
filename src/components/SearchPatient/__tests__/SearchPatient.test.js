import { shallow, mount } from "enzyme";
import React from "react";
import SearchPatient from "./../SearchPatient";
import SearchIcon from "@material-ui/icons/Search";

describe("SearchPatient", () => {
  const mock = jest.fn();
  let props = {
    onSearch: mock,
    patientId: "1",
    loading: false
  };

  it("should render properly", () => {
    const wrapper = shallow(<SearchPatient {...props} />);
    expect(wrapper.find(".icon-button").length).toEqual(1);
    expect(wrapper.find(SearchIcon).length).toBe(1);
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper
        .find("#search-field")
        .at(0)
        .props()["disabled"]
    ).toBe(false);
  });

  it("should disable input field to edit, while loading the patient id", () => {
    props.loading = true;
    const wrapper = mount(<SearchPatient {...props} />);
    expect(
      wrapper
        .find("#search-field")
        .at(0)
        .props()["disabled"]
    ).toBe(true);

    expect(
      wrapper
        .find("#search-field")
        .at(0)
        .props()["value"]
    ).toBe("Looking for ");
    // expect(wrapper.find("#loader").length).toBe(1);
  });

  it("should set input box value back to patient id after loading is done", () => {
    const wrapper = mount(<SearchPatient {...props} />);
    // const setTextInput = jest.fn();
    expect(
      wrapper
        .find("#search-field")
        .at(0)
        .props()["disabled"]
    ).toBe(true);
    // expect(setTextInput).toHaveBeenCalledWith("1");
    expect(
      wrapper
        .find("#search-field")
        .at(0)
        .props()["value"]
    ).toBe("Looking for ");
  });
});
