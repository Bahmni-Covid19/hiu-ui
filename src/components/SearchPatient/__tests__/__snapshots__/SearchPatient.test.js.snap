// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPatient should render properly 1`] = `
<styled.div>
  <div
    className="search-bar"
  >
    <WithStyles(ForwardRef(TextField))
      InputProps={
        Object {
          "endAdornment": <ForwardRef(WithStyles)
            position="end"
          >
            @ncg
          </ForwardRef(WithStyles)>,
        }
      }
      disabled={false}
      error={false}
      helperText=""
      id="search-field"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="search"
      value=""
    />
    <WithStyles(ForwardRef(IconButton))
      aria-label="search"
      className="icon-button"
      disabled={true}
      onClick={[Function]}
      theme="primary"
      type="button"
    >
      <SearchIcon
        className="search-button"
        id="search-button"
        theme="primary"
        variant="outlined"
      />
    </WithStyles(ForwardRef(IconButton))>
  </div>
</styled.div>
`;
